From mboxrd@z Thu Jan  1 00:00:00 1970
Received: from mail-oa1-f54.google.com (mail-oa1-f54.google.com [209.85.160.54])
	(using TLSv1.2 with cipher ECDHE-RSA-AES128-GCM-SHA256 (128/128 bits))
	(No client certificate requested)
	by smtp.subspace.kernel.org (Postfix) with ESMTPS id A44AA217F31
	for <linux-wireless@vger.kernel.org>; Thu, 19 Feb 2026 00:40:29 +0000 (UTC)
Authentication-Results: smtp.subspace.kernel.org; arc=none smtp.client-ip=209.85.160.54
ARC-Seal:i=1; a=rsa-sha256; d=subspace.kernel.org; s=arc-20240116;
	t=1771461630; cv=none; b=iOzb55M+XUcfOdDxOGIkCpi3Rw9TNQPZn+Hg8CCZagaLBF4ilWr3PJj2I09a+c1QES8dOhdJW+X4UN+MjRG5vWqBgsbNTIil+DNW5wg0P/ies+I6bLWcoID2BdvoMMEaYEDjk5p5j2ILAVIH31TMdzS53taweYLxl6cdRKBLuP0=
ARC-Message-Signature:i=1; a=rsa-sha256; d=subspace.kernel.org;
	s=arc-20240116; t=1771461630; c=relaxed/simple;
	bh=T5b7BWZwpQvP/l0ifB56jc3DgOo8jjpnV//rdmhZRHQ=;
	h=From:To:Cc:Subject:Date:Message-ID:In-Reply-To:References:
	 MIME-Version; b=dfCcTG1nyk0POv0ANtJaFyOcwZknKGsjx/hQyPfACA7T3vf3vILQJZQJEAcgiwozzUy4Q0ggRdyOUoiXuN72yKmXL2u6vkU1oF5yLQD26mz7rKINoxsL2CGthrP4ndXnGhuG0Hf2bKNWTPeqhglFhnjsmEFDIjVKDbRx/vPCGkM=
ARC-Authentication-Results:i=1; smtp.subspace.kernel.org; dmarc=fail (p=quarantine dis=none) header.from=kernel.org; spf=pass smtp.mailfrom=gmail.com; arc=none smtp.client-ip=209.85.160.54
Authentication-Results: smtp.subspace.kernel.org; dmarc=fail (p=quarantine dis=none) header.from=kernel.org
Authentication-Results: smtp.subspace.kernel.org; spf=pass smtp.mailfrom=gmail.com
Received: by mail-oa1-f54.google.com with SMTP id 586e51a60fabf-404254ffe8aso303659fac.0
        for <linux-wireless@vger.kernel.org>; Wed, 18 Feb 2026 16:40:29 -0800 (PST)
X-Google-DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
        d=1e100.net; s=20230601; t=1771461628; x=1772066428;
        h=content-transfer-encoding:mime-version:references:in-reply-to
         :message-id:date:subject:cc:to:from:x-gm-gg:x-gm-message-state:from
         :to:cc:subject:date:message-id:reply-to;
        bh=R8frkFgL2LcQ46RFOXtOLVg7ByeEmIT9eU+ttG4MBVU=;
        b=AeKinqNUL0dMl3QlFbxYpgew1Ljz2ULgzcZ779He97xWI26lnjj2z0QhExEXVP3UHL
         nRyksBJm5XlW3k4vkN0fgbWQ8tw4HAIjtbWdBcaQG+ErO0DVJIFBoRwmvHZpyFk072OM
         taUppZlQ9ZFyqF+4Mchhlf72GbWXvmwLhS7h1k9MOCxthK6YFVXie9+gqseRFSds8iQb
         7aFo+bLTrgzP3oJtM/JcHya5KIgWc/A6X4vYbpV3r4TdJCGNlkymDqstmiMHmW8txWFI
         UBKy/DVZ+Tcg/giOfhfo1h+PTFZ1CuBebxNAihbDylCzAoVIEFBCxIbrud78CV2JdNxA
         IK3Q==
X-Gm-Message-State: AOJu0Yz35Dc6U9H7D99fLdCpz/bu9oXQ7tZHMIaDNTQ8QDjz4XfaoBVQ
	h8KneHFC3pcP3QtWZWA/B/tyU8zl1QaeUt07vg6vm6DK2VR07wNCj92f
X-Gm-Gg: AZuq6aIXbhAAywH6BX1bx6vNza11twVLrpn9ZCrYnEBJ8jIZ0L59sBk/hTBvt/g7CAL
	KuoTlmyQazg7ny+ENOYmfNUXmYD9h/9G92cP3b+te9cqeJZfHUxE0iW8hAqImSFhzbmgGMt8wEh
	Gf6vG5g/x531vhKsVaPD4utEYrYBj3OOyX859/7yl+CeI4LrhB5c+tuwYrxHH4SPC+ueqbmmqGw
	oaV3cKgP8z+f2v0PlF3xF1KXx7MV+44aLuN1v1SLsZRLWLr2lfYFa79l9YMd7VCKC22kmacu0UW
	PNrh+48rVh4huIJbPG1/k7C7h/dp/O2+IYObyjcOD9l8STzyUYBjoYoYSWzTf7EKTmDk5GdJoDz
	eABIhZMCrY2Ht6i6bnUlgTRkqazwPDhT1hOsrAQVUQMgkswxHfL8DCLzlJchIHp5tdk7rgDA8iT
	/H4Ch3t+tG8+dJ8HZZKhU6f9sM4WkPmaGHK3oel3dh0xDbrQY=
X-Received: by 2002:a05:6870:37cc:b0:413:9c82:8742 with SMTP id 586e51a60fabf-415458deaeemr52956fac.27.1771461628511;
        Wed, 18 Feb 2026 16:40:28 -0800 (PST)
Received: from sean-HP-EliteBook-830-G6.lan ([207.191.35.252])
        by smtp.gmail.com with ESMTPSA id 586e51a60fabf-40eaee45d12sm23477659fac.1.2026.02.18.16.40.27
        (version=TLS1_3 cipher=TLS_AES_256_GCM_SHA384 bits=256/256);
        Wed, 18 Feb 2026 16:40:28 -0800 (PST)
From: sean.wang@kernel.org
To: nbd@nbd.name,
	lorenzo.bianconi@redhat.com
Cc: linux-wireless@vger.kernel.org,
	linux-mediatek@lists.infradead.org,
	Sean Wang <sean.wang@mediatek.com>,
	Xiong Huang <xiong.huang@mediatek.com>
Subject: [PATCH 10/11] wifi: mt76: mt7921: add MT7902 PCIe device support
Date: Wed, 18 Feb 2026 18:40:06 -0600
Message-ID: <20260219004007.19733-10-sean.wang@kernel.org>
X-Mailer: git-send-email 2.43.0
In-Reply-To: <20260219004007.19733-1-sean.wang@kernel.org>
References: <20260219004007.19733-1-sean.wang@kernel.org>
Precedence: bulk
X-Mailing-List: linux-wireless@vger.kernel.org
List-Id: <linux-wireless.vger.kernel.org>
List-Subscribe: <mailto:linux-wireless+subscribe@vger.kernel.org>
List-Unsubscribe: <mailto:linux-wireless+unsubscribe@vger.kernel.org>
MIME-Version: 1.0
Content-Transfer-Encoding: 8bit

From: Sean Wang <sean.wang@mediatek.com>

Register the MT7902 PCI device ID in the mt7921 driver and add its
corresponding firmware and ROM patch names.

Co-developed-by: Xiong Huang <xiong.huang@mediatek.com>
Signed-off-by: Xiong Huang <xiong.huang@mediatek.com>
Signed-off-by: Sean Wang <sean.wang@mediatek.com>
---
 drivers/net/wireless/mediatek/mt76/mt7921/pci.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/drivers/net/wireless/mediatek/mt76/mt7921/pci.c b/drivers/net/wireless/mediatek/mt76/mt7921/pci.c
index 6bb3c6a1cf6a..7a790ddf43bb 100644
--- a/drivers/net/wireless/mediatek/mt76/mt7921/pci.c
+++ b/drivers/net/wireless/mediatek/mt76/mt7921/pci.c
@@ -26,6 +26,8 @@ static const struct pci_device_id mt7921_pci_device_table[] = {
 		.driver_data = (kernel_ulong_t)MT7922_FIRMWARE_WM },
 	{ PCI_DEVICE(PCI_VENDOR_ID_MEDIATEK, 0x7920),
 		.driver_data = (kernel_ulong_t)MT7920_FIRMWARE_WM },
+	{ PCI_DEVICE(PCI_VENDOR_ID_MEDIATEK, 0x7902),
+		.driver_data = (kernel_ulong_t)MT7902_FIRMWARE_WM },
 	{ },
 };
 
@@ -617,6 +619,8 @@ MODULE_FIRMWARE(MT7921_FIRMWARE_WM);
 MODULE_FIRMWARE(MT7921_ROM_PATCH);
 MODULE_FIRMWARE(MT7922_FIRMWARE_WM);
 MODULE_FIRMWARE(MT7922_ROM_PATCH);
+MODULE_FIRMWARE(MT7902_FIRMWARE_WM);
+MODULE_FIRMWARE(MT7902_ROM_PATCH);
 MODULE_AUTHOR("Sean Wang <sean.wang@mediatek.com>");
 MODULE_AUTHOR("Lorenzo Bianconi <lorenzo@kernel.org>");
 MODULE_DESCRIPTION("MediaTek MT7921E (PCIe) wireless driver");
-- 
2.43.0


